api/nodes/views.py

```diff
@@ -1151,6 +1153,78 @@ def get_object(self):
         return fobj
 
 
+class NodeGroupsList(JSONAPIBaseView, generics.ListCreateAPIView, NodeMixin, ListFilterMixin):
+    """ The documentation for this endpoint can be found [here](https://developer.osf.io/#operation/nodes_groups_list)
+
+    """
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+        AdminOrPublic,
+        base_permissions.TokenHasScope,
+    )
+
+    serializer_class = NodeGroupsSerializer
+
+    required_read_scopes = [CoreScopes.NODE_OSF_GROUPS_READ]
+    required_write_scopes = [CoreScopes.NODE_OSF_GROUPS_WRITE]
+    view_category = 'nodes'
+    view_name = 'node-groups'
+
+    def get_default_queryset(self):
+        return self.get_node().osf_groups
+
+    def get_queryset(self):
+        return self.get_queryset_from_request()
+
+    # overrides FilterMixin
+    def build_query_from_field(self, field_name, operation):
+        if field_name == 'permission':
+            node = self.get_node()
+            osf_groups = node.osf_groups
+            group_ids = [
```

Comment:
This logic felt necessarily complicated, and I spent awhile trying to make it nicer. Can anyone point out a better way to get all of the groups granted a given permission on the specified node?